### DISCUSSION TITLE
Response to: "We can't help but feel genuinely disappointed ..."

### DISCUSSION THREAD

**Discussion by janpio** [prisma/prisma#21099] [Q&A]
@leonard-henriquez in https://github.com/prisma/prisma/issues/1644#issuecomment-1720969416_

> We can't help but feel genuinely disappointed to see Prisma's focus shift away from its foundational open-source ORM to building new paid cloud tools. While we understand that business goals evolve, this shift seems like a missed opportunity to foster deeper community engagement. If improving your ORM is no longer your primary focus, **we'd love to see the door open for more community contributions in this area**.
> 
> It's been a disheartening year for the open-source community, with events like Terraform changing its license only adding to our concerns. The recent developments at Prisma further deepen those concerns. We sincerely hope there's room to reconsider this approach for the betterment of all involved. 🥲

Uhhh, I did not write any of that in my message:

> We don't communicate concrete dates, so I also can't promise we will get to this in the next 6 months. If you absolutely need this, your better choice is indeed to use another tool. We hope to win you back when we implement this in the future.

Prisma's focus has not shifted away from the ORM. Nothing really changed here at all. The cloud products are built by a different team in the company. I think us working on the ORM have actually been shipping more, and more impactful features and improvements, than ever before. We will continue doing so.

Someone asked for a hard date commitment, "Will this feature make it in the priorities in the coming 6 months ?", and I am not comfortable giving those as the Product Manager of Prisma ORM. We never do. Not 3 years ago, not today. So I suggest you sould not read more into this statement than what it says.

**Comment by leonard-henriquez** [comment]
@janpio 

Regarding the ETA:
> Someone asked for a hard date commitment, "Will this feature make it in the priorities in the coming 6 months ?", and I am not comfortable giving those as the Product Manager of Prisma ORM. We never do. Not 3 years ago, not today. So I suggest you sould not read more into this statement than what it says.

Someone mentioned priority, but I'm not familiar with that person. You may want to tag them if you intend to respond. However, based on the overall sentiment from the other comments on this issue (https://github.com/prisma/prisma/issues/1644), it seems like people are more interested in transparency regarding this issue's place on the roadmap than in a "hard date commitment".

---

Regarding Prisma's focus on the ORM:
> Prisma's focus has not shifted away from the ORM. Nothing really changed here at all. The cloud products are built by a different team in the company. I think us working on the ORM have actually been shipping more, and more impactful features and improvements, than ever before. We will continue doing so.

I get that Polymorphic Associations might not be Prisma's top priority. However, based on the considerable interest in this thread it's clear that many users, including my own team, consider this a fundamental feature for an ORM. We've even offered to contribute code to help speed up the process. That said, my takeaway from your last message was that you're not open to external contributions on this topic. Did I misunderstand something?

> Straightforward answer is that you can't. No one here is available to jump into any discussions right now, no one is available to review any PRs that try to partially implement this, and no one is available right now to start the design or implementation work. (If you can help out in other issues to make sure the problem is described, a minimal reproduction is available and so on that theoretically saves us some time - but that only very indirectly translates to benefits for this set of issues.)
> From @janpio in this comment: https://github.com/prisma/prisma/issues/1644#issuecomment-1552242312

**Reply by janpio** [reply]
Again, two very different things in my opinion: Us not being a community-driven project, or being set up to mostly support community contributions is very different from "Prisma's focus shift away from its foundational open-source ORM". Prisma ORM has always been driven by the company Prisma, with people paid for working on the product.

We are as focussed on the ORM as we were in the last 4 years (since I am at this company).
We did not change the way we work on Prisma ORM, which always was driven by the company with occasional community contributions.

That said, my wording was not the best in that comment. Sorry.

If you have a proposal how to design [Polymorphic Associations](https://github.com/prisma/prisma/issues/1644) in Prisma, let us know. Post a gist or long comment that explains how it would work, what syntax additions are required, what the queries under the hood would look like. We will not jump on it _instantly_ - as we are already busy with things that we have planned internally and want to ship with our next release(s). But that would probably be how to best influence the progress here.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
